src/deferred.js

```diff
@@ -212,20 +215,26 @@ jQuery.extend( {
 										}
 									};
 
+							processQueue.push( process );
+
 							// Support: Promises/A+ section 2.3.3.3.1
 							// https://promisesaplus.com/#point-57
 							// Re-resolve promises immediately to dodge false rejection from
 							// subsequent errors
 							if ( depth ) {
-								process();
+								deferredTick();
```

Comment:
Sorry, I think the case I was looking for was `deferred.then(fn1).notify(fn2)` since I think that's the only time `special` comes into play? This definitely could use a @gibson042 look.